<a href="https://colab.research.google.com/github/thisaraniNJ/Aircraft_Damage_Prediction/blob/main/Aircraft_Damage_Prediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Importing Necessary Libraries

In [11]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from warnings import filterwarnings
filterwarnings('ignore')

pd.set_option('display.max_columns', None) # To represent all the columns in the dataset

# Loading Data

In [12]:
Aviation = pd.read_csv('/content/drive/MyDrive/My ML Projects/AviationData.csv', encoding='mac_roman')

In [13]:
Aviation.head()

Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country   Latitude  Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN        NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN        NaN          NaN   
2    Saltville, VA  United States  36.922223 -81.878056          NaN   
3       EUREKA, CA  United States        NaN        NaN          NaN   
4       Canton, OH  United States        NaN        NaN          NaN   

  Airport.Name Injury.Severity Aircraft.damage Aircraft.Category  \
0          NaN        Fatal(2)       Destroyed               NaN   
1          NaN        Fatal(4)       Destroyed               NaN   
2          NaN        Fatal(3)       Destroyed               NaN   
3          NaN        Fatal(2)       Destroyed               NaN   
4          NaN        Fatal(1)       Destroyed               NaN   

  Registration.Number      Make     Model Amateur.Built  Number.of.Engines  \
0              NC6404   Stinson     108-3            No                1.0   
1              N5069P     Piper  PA24-180            No                1.0   
2              N5142R    Cessna      172M            No                1.0   
3              N1168J  Rockwell       112            No                1.0   
4               N15NY    Cessna       501            No                NaN   

     Engine.Type FAR.Description Schedule Purpose.of.flight Air.carrier  \
0  Reciprocating             NaN      NaN          Personal         NaN   
1  Reciprocating             NaN      NaN          Personal         NaN   
2  Reciprocating             NaN      NaN          Personal         NaN   
3  Reciprocating             NaN      NaN          Personal         NaN   
4            NaN             NaN      NaN          Personal         NaN   

   Total.Fatal.Injuries  Total.Serious.Injuries  Total.Minor.Injuries  \
0                   2.0                     0.0                   0.0   
1                   4.0                     0.0                   0.0   
2                   3.0                     NaN                   NaN   
3                   2.0                     0.0                   0.0   
4                   1.0                     2.0                   NaN   

   Total.Uninjured Weather.Condition Broad.phase.of.flight   Report.Status  \
0              0.0               UNK                Cruise  Probable Cause   
1              0.0               UNK               Unknown  Probable Cause   
2              NaN               IMC                Cruise  Probable Cause   
3              0.0               IMC                Cruise  Probable Cause   
4              0.0               VMC              Approach  Probable Cause   

  Publication.Date  
0              NaN  
1       19-09-1996  
2       26-02-2007  
3       12-09-2000  
4       16-04-1980

In [14]:
Aviation.shape

(87282, 31)

There are 87282 flights have been captured to make a prediction

# Data Description

Non-Null Values, Datatypes, % of null values

In [20]:
info_dict = {
    "Column": Aviation.columns.tolist(),
    "Non-Null Count" : Aviation.count(),
    "Dtypes": Aviation.dtypes.tolist(),
    "Null Precentage":(Aviation.shape[0]-Aviation.count())/Aviation.shape[0]*100
}

Description = pd.DataFrame(info_dict).reset_index(drop=True)


In [21]:
Description

Column  Non-Null Count   Dtypes  Null Precentage
0                 Event.Id           87282   object         0.000000
1       Investigation.Type           87282   object         0.000000
2          Accident.Number           87282   object         0.000000
3               Event.Date           87282   object         0.000000
4                 Location           87230   object         0.059577
5                  Country           87056   object         0.258931
6                 Latitude           33073   object        62.107880
7                Longitude           33064   object        62.118192
8             Airport.Code           49217   object        43.611512
9             Airport.Name           51870   object        40.571939
10         Injury.Severity           86399   object         1.011663
11         Aircraft.damage           84272   object         3.448592
12       Aircraft.Category           30783   object        64.731560
13     Registration.Number           85900   object         1.583373
14                    Make           87219   object         0.072180
15                   Model           87190   object         0.105405
16           Amateur.Built           87180   object         0.116863
17       Number.of.Engines           81533  float64         6.586696
18             Engine.Type           81372   object         6.771156
19         FAR.Description           30458   object        65.103916
20                Schedule           12330   object        85.873376
21       Purpose.of.flight           81493   object         6.632524
22             Air.carrier           15971   object        81.701840
23    Total.Fatal.Injuries           75881  float64        13.062258
24  Total.Serious.Injuries           74772  float64        14.332852
25    Total.Minor.Injuries           75349  float64        13.671777
26         Total.Uninjured           81370  float64         6.773447
27       Weather.Condition           83304   object         4.557641
28   Broad.phase.of.flight           61724   object        29.282097
29           Report.Status           82067   object         5.974886
30        Publication.Date           73629   object        15.642400

Statistical Summary

In [22]:
Aviation.describe()

Number.of.Engines  Total.Fatal.Injuries  Total.Serious.Injuries  \
count       81533.000000          75881.000000            74772.000000   
mean            1.147131              0.652772                0.280921   
std             0.447085              5.521338                1.557476   
min             0.000000              0.000000                0.000000   
25%             1.000000              0.000000                0.000000   
50%             1.000000              0.000000                0.000000   
75%             1.000000              0.000000                0.000000   
max             8.000000            349.000000              161.000000   

       Total.Minor.Injuries  Total.Uninjured  
count          75349.000000     81370.000000  
mean               0.360814         5.311847  
std                2.257721        27.899156  
min                0.000000         0.000000  
25%                0.000000         0.000000  
50%                0.000000         1.000000  
75%                0.000000         2.000000  
max              380.000000       699.000000

In [24]:
Aviation.describe(include='object', exclude='number')

Event.Id Investigation.Type Accident.Number  Event.Date  \
count            87282              87282           87282       87282   
unique           86370                  2           87282       14435   
top     20001212X19172           Accident      SEA87LA080  1984-06-30   
freq                 3              83536               1          25   

             Location        Country Latitude Longitude Airport.Code  \
count           87230          87056    33073     33064        49217   
unique          27171            210    24752     26206        10252   
top     ANCHORAGE, AK  United States  332739N  0112457W         NONE   
freq              434          80998       19        22         1486   

       Airport.Name Injury.Severity Aircraft.damage Aircraft.Category  \
count         51870           86399           84272             30783   
unique        24545             107               4                12   
top         Private       Non-Fatal     Substantial          Airplane   
freq            232           66561           62975             26343   

       Registration.Number    Make  Model Amateur.Built    Engine.Type  \
count                85900   87219  87190         87180          81372   
unique               77696    8045  12185             4             10   
top                   NONE  Cessna    152            No  Reciprocating   
freq                   344   22227   2350         58165          69159   

       FAR.Description Schedule Purpose.of.flight Air.carrier  \
count            30458    12330             81493       15971   
unique              31        3                28       13079   
top                091     NSCH          Personal       Pilot   
freq             17131     4284             48655         246   

       Weather.Condition Broad.phase.of.flight   Report.Status  \
count              83304                 61724           82067   
unique                 4                    12           16666   
top                  VMC               Landing  Probable Cause   
freq               76251                 15428           61754   

       Publication.Date  
count             73629  
unique             2686  
top          25-09-2020  
freq              17019